.github/workflows/auto-update-upgrade.yml

```diff
@@ -0,0 +1,32 @@
+# When new release is created this workflow will be triggered and will do 3 things:
+# 1) it will create a directory with an empty upgrade handler in app/upgrades folder
+# 2) will increase an E2E_UPGRADE_VERSION variable in Makefile
+# 3) create a pull request with these changes to main 
+
+name: On Release Auto Upgrade
+
+on:
+  release: 
+    types: [created]
```

Comment:
Does this only run on a major upgrade only?

We would like to avoid this running for minor/patch upgrades

Reply:
hello @p0mvn. So basically what you mean is that it should run when in version vX.Y.Z of osmosis only X changes, for example from 12 to 13, right?

Reply:
if so, then unfortunately this script will run when any of X Y or Z changes

but this can be addressed

Reply:
also, are you sure you want this action to be run when a release is created and not published?

Reply:
I think when the release is published is preferable

Reply:
got it!